alibi_detect/utils/keops/tests/test_kernels_keops.py

```diff
@@ -0,0 +1,45 @@
+from itertools import product
+import numpy as np
+from alibi_detect.utils.frameworks import has_keops
+import pytest
+import torch
+if has_keops:
+    from pykeops.torch import LazyTensor
+    from alibi_detect.utils.keops import GaussianRBF
+
+sigma = [None, np.array([1.]), np.array([1., 2.])]
+n_features = [5, 10]
+n_instances = [(100, 100), (100, 75)]
+trainable = [True, False]
+tests_gk = list(product(sigma, n_features, n_instances, trainable))
+n_tests_gk = len(tests_gk)
+
+
+@pytest.fixture
+def gaussian_kernel_params(request):
+    return tests_gk[request.param]
+
+
+@pytest.mark.skipif(not has_keops, reason='Skipping since pykeops is not installed.')
+@pytest.mark.parametrize('gaussian_kernel_params', list(range(n_tests_gk)), indirect=True)
+def test_gaussian_kernel(gaussian_kernel_params):
+    sigma, n_features, n_instances, trainable = gaussian_kernel_params
+    xshape, yshape = (n_instances[0], n_features), (n_instances[1], n_features)
+
+    print(sigma, xshape, yshape, trainable)
+
+    sigma = sigma if sigma is None else torch.from_numpy(sigma).float()
+    x = torch.from_numpy(np.random.random(xshape)).float()
+    y = torch.from_numpy(np.random.random(yshape)).float()
+
+    kernel = GaussianRBF(sigma=sigma, trainable=trainable)
+    infer_sigma = True if sigma is None else False
+    if trainable and infer_sigma:
+        with pytest.raises(Exception):
```

Comment:
`Exception` -> `ValueError` for tighter control. I note that the same could be done for the `tensorflow` and `torch` tests.